AK/Error.cpp

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2023, kleines Filmröllchen <filmroellchen@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "Error.h"
+#include "String.h"
+
+namespace AK {
+
+Error::Error(Error&& other)
+{
+    swap(this->m_syscall, other.m_syscall);
+    swap(this->m_code, other.m_code);
+#if !defined(KERNEL)
+    if (other.m_string_literal.has<StringView>())
+        this->m_string_literal = move(other.m_string_literal.get<StringView>());
+    else if (other.m_string_literal.get<String const*>() != nullptr) {
+        this->m_string_literal = move(other.m_string_literal.get<String const*>());
+        // Make sure the other Error doesn't destruct the string!
+        other.m_string_literal.get<String const*>() = nullptr;
+    }
```

Comment:
If one branch in an if/else-if chain requires braces, all chains should have braces.